processor/src/main/java/org/mapstruct/ap/services/Services.java

```diff
@@ -0,0 +1,77 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.services;
+
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.WeakHashMap;
+
+import org.mapstruct.spi.AccessorNamingStrategy;
+
+/**
+ * A simple locator for SPI implementations.
+ *
+ * @author Christian Schuster
+ */
+public class Services {
+
+    private static final Map<Class<?>, Object> CACHE = new WeakHashMap<Class<?>, Object>();
```

Comment:
You're right. I was just going for a "normal" cache and didn't take the short-livedness (is that even a word?) of the classloader into account. I'll change it.